src/framework/ui/view/keynavigationsubsection.h

```diff
@@ -32,18 +32,30 @@ class KeyNavigationControl;
 class KeyNavigationSubSection : public AbstractKeyNavigation, public IKeyNavigationSubSection, public async::Asyncable
 {
     Q_OBJECT
-    Q_PROPERTY(KeyNavigationSection * section READ section WRITE setSection)
+    Q_PROPERTY(KeyNavigationSection * section READ section WRITE setSection NOTIFY sectionChanged)
+    Q_PROPERTY(QmlDirection direction READ direction_property WRITE setDirection NOTIFY directionChanged)
 
 public:
     explicit KeyNavigationSubSection(QObject* parent = nullptr);
     ~KeyNavigationSubSection() override;
 
+    //! NOTE Must be sync with IKeyNavigationSubSection::Direction
```

Comment:
I afraid that we'll forget to add new values here

Reply:
added comment in both place